src/Components/Aspire.Microsoft.Azure.Cosmos/AspireMicrosoftAzureCosmosExtensions.cs

```diff
@@ -101,48 +236,42 @@ private static void AddAzureCosmosClient(
             });
         }
 
+        configureClientOptions?.Invoke(clientOptions);
+
         if (CosmosUtils.IsEmulatorConnectionString(settings.ConnectionString))
         {
             clientOptions.ConnectionMode = ConnectionMode.Gateway;
             clientOptions.LimitToEndpoint = true;
         }
 
-        configureClientOptions?.Invoke(clientOptions);
```

Comment:
Why did we change the order of the calls here? The intention is to give the developer the last chance to change options. So by moving this call above the IsEmulatorConnectionString check, we are always going to overwrite the above 2 options.

Reply:
This wasn't intentionally. I'll try to figure out a way for us to test the ordering behavior here. 🤔